components/once/once.c

```diff
@@ -0,0 +1,30 @@
+#include "once.h"
+
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <uv.h>
+
+int datadog_php_once_ctor(struct datadog_php_once_s *once) {
+  atomic_store(&once->initialized, false);
+  return uv_mutex_init(&once->mutex);
+}
+
+void datadog_php_once_dtor(struct datadog_php_once_s *once) {
+  atomic_store(&once->initialized, true);
```

Comment:
Setting `true` here seems strange. It does guarantee that you will never call the init_routine again, though the boolean captures several states with this value.

Reply:
This is a destructor, so doing pretty much anything would be legitimate here for the var `initialized` -- if they use it again in any way except to `ctor` it, then it's invalid usage. Setting `initialized` to `true` in the destructor provides some amount of safety in the event of invalid usage, but that's a "best effort" kind of thing, I think?

Reply:
I think it is OK, the only benefit of having a separate value (using an enum) is when you are investigating a possible crash and want to figure out which state things are in.